larray/core/session.py

```diff
@@ -303,22 +321,15 @@ def __setitem__(self, key, value):
     def __delitem__(self, key):
         del self._objects[key]
 
-    # TODO: add a a meta property when Python 2.7 will be dropped
     def __getattr__(self, key):
-        if key == 'meta':
-            return self._meta
-        elif key in self._objects:
+        if key in self._objects:
             return self._objects[key]
         else:
             raise AttributeError("'{}' object has no attribute '{}'".format(self.__class__.__name__, key))
 
-    # TODO: implement meta.setter when Python 2.7 will be dropped
     def __setattr__(self, key, value):
         if key == 'meta':
-            if not isinstance(value, (list, dict, OrderedDict, Metadata)):
-                raise TypeError("Expected list of pairs or dict or OrderedDict or Metadata object "
-                                "instead of {}".format(type(value).__name__))
-            object.__setattr__(self, '_meta', value if isinstance(value, Metadata) else Metadata(value))
+            super().__setattr__(key, value)
```

Comment:
I don't think this code is necessary/will ever be executed as I **think** `__setattr__` will only be called if there is no method nor property with that name on the class. It is worth double checking though.

Reply:
I tried but it seems that line is necessary : https://stackoverflow.com/a/15751159

Reply:
Yuk! I didn't know (or forgot) about the asymmetry between `__getattr__` and `__setattr__`. Please add a comment about this.